core/src/main/scala/com/metamx/tranquility/druid/IndexService.scala

```diff
@@ -166,6 +188,38 @@ object IndexService
   type TaskId = String
   type TaskPayload = Dict
 
+  case class TaskHostPort(host: String, port: Int)
+  {
+    def isBound = host != null && host.nonEmpty && port > 0
+
+    def toAddr: Addr = {
+      if (!isBound) {
+        Addr.Neg
+      } else {
+        Addr.Bound(new InetSocketAddress(host, port))
+      }
+    }
+  }
+
+  object TaskHostPort
+  {
+    def unknown = TaskHostPort("", -1)
```

Comment:
-1 is meant to indicate "unknown" and is also what the overlord would report if the port is unknown.